### PR TITLE
build UMD with same compiler used to compile metal and remove clang 6 as a dependency

### PR DESCRIPTION
**Pull request by TT-billteng**
UMD should work with the same compiler we use for Metal

### CODE REVIEW

File: INSTALLING.md [hunk 1]
```diff
@@ -124,7 +124,7 @@ Please follow the next additional steps if you want to contribute to the codebas
 1. Install dependencies
 
 ```sh
-sudo apt install clang-6.0=1:6.0.1-14 git git-lfs cmake=3.16.3-1ubuntu1.20.04.1 pandoc libtbb-dev libcapstone-dev pkg-config ninja-build patchelf
+sudo apt install git git-lfs cmake=3.16.3-1ubuntu1.20.04.1 pandoc libtbb-dev libcapstone-dev pkg-config ninja-build patchelf
```
Comment (ayerofieiev-tt) [inline INSTALLING.md:127]: should instead install new version?

**Review by ayerofieiev-tt — APPROVED** [review]

**Comment by davorchap** [comment]
Do we need to keep clang-6 for the Versim build?

That could be a separate build target

**Review by vtangTT — APPROVED** [review]

**Comment by TT-billteng** [comment]
> Do we need to keep clang-6 for the Versim build?
> 
> That could be a separate build target

Yes should be a separate target, but the libraries could be generated with a newer clang from what I've heard

File: INSTALLING.md (hunk 1 above)
Comment (TT-billteng) [reply INSTALLING.md:127]: new version (clang-17) is in the same file

**Review by tt-rkim — APPROVED** [review]
just gotta make sure we delete it from the install deps as well from MIW

**Pull request merged**